<h3 style=" color: lightblue;">1-L'IMPORTATION DES BIBLIOTHÈQUES : </h3>

In [ ]:
import pandas as pd          # Manipulation des données (DataFrame)
import numpy as np           # Calculs numériques
import math                  # Fonctions mathématiques de base
import matplotlib.pyplot as plt  # Visualisation (histogrammes, courbes)
from scipy import stats      # Tests statistiques (ttest, distributions)
import seaborn as sns       

<h3 style=" color: lightblue;">2-L'IMPORTATION DES DONNEES :</h3>

In [8]:
import pandas as pd
# Charger le dataset complet
dataset_path = 'C:\\Users\\wafae\\Desktop\\sleep_health_stat_project\\data\\Sleep_health_and_lifestyle_dataset.csv'

data = pd.read_csv(dataset_path)

# Extraire uniquement la colonne Sleep Duration
sleep_duration = data["Sleep Duration"]
# Vérifier les premières lignes
# sleep_duration.head()
# Sauvegarder la colonne Sleep Duration dans un fichier texte
sleep_duration.to_csv("../data/sleep_duration.txt", index=False, header=True)
# Vérifier les premières lignes
sleep_duration.head()

0    6.1
1    6.2
2    6.2
3    5.9
4    5.9
Name: Sleep Duration, dtype: float64

<div style="background-color: #f0f8ff; padding: 10px; border-left: 5px solid #1e90ff; border-radius: 5px; color: black;">
<b>Explication :</b><br>
La <b>valeur minimale</b> et la <b>valeur maximale</b> de Sleep Duration indiquent les bornes extrêmes de l'échantillon.<br>
- La valeur minimale (5.8 heures) correspond à la personne dormant le moins.<br>
- La valeur maximale (8.5 heures) correspond à la personne dormant le plus.<br>
Ces valeurs sont réalistes pour un échantillon adulte.
</div>


<h3 style="color: lightblue;">3 - CALCUL DES VALEURS MINIMALES ET MAXIMALES DE L'ÉCHANTILLON :</h3>


In [9]:
# Valeur minimale et maximale de Sleep Duration
# calcule la plus petite valeur
min_value = round(sleep_duration.min(), 2)
# calcule la plus grande valeur
max_value = round(sleep_duration.max(), 2)
# round(..., 2) → arrondit a 2 decimales pour plus de lisibilite
print(f"Valeur minimale : {min_value} heures")
print(f"Valeur maximale : {max_value} heures")

Valeur minimale : 5.8 heures
Valeur maximale : 8.5 heures


<h3 style=" color: lightblue;">4-CALCULE LA MOYENNE DE L'ECHANTILLON :</h3>


In [10]:
# calcul de moyenne de sleep duration
mean_sleep = round(sleep_duration.mean(), 2)
print(f"Moyenne de l'échantillon : {mean_sleep} heures")

Moyenne de l'échantillon : 7.13 heures


<div style="background-color: #f0f8ff; padding: 10px; border-left: 5px solid #1e90ff; border-radius: 5px; color: black;">
<b>Explication :</b><br>
La <b>moyenne</b> de l'échantillon (7.13 heures) indique la durée de sommeil moyenne pour toutes les personnes de l'échantillon.<br>
C'est une mesure de tendance centrale qui résume bien la variable Sleep Duration.
</div>


<h3 style=" color: lightblue;">5-CALCULE LA TAILLE DE L'ECHANTILLON :</h3>

In [11]:
n = len(sleep_duration)
print(f"Taille de l'échantillon : {n}")

Taille de l'échantillon : 374


<div style="background-color: #f0f8ff; padding: 10px; border-left: 5px solid #1e90ff; border-radius: 5px; color: black;">
<b>Explication :</b><br>
La <b>taille de l'échantillon</b> (374) correspond au nombre total de personnes observées.<br>
Même si nous avons le dataset complet, on l'appelle échantillon car il représente une partie d'une population plus large.
</div>


<h3 style=" color: lightblue;">6-CALCULE LA VARIANCE DE L'ECHANTILLON :</h3>

In [12]:
variance = sleep_duration.var(ddof=1)
print(f"Variance : {round(variance,2)}")


Variance : 0.63


<div style="background-color: #f0f8ff; padding: 10px; border-left: 5px solid #1e90ff; border-radius: 5px; color: black;">
<b>Explication :</b><br>
La <b>variance</b> (0.63) mesure la dispersion des durées de sommeil autour de la moyenne.<br>
Elle indique à quel point les valeurs s'éloignent de la moyenne (en heures²).<br>
Plus la variance est faible, plus les durées sont proches de la moyenne.
</div>


<h3 style=" color: lightblue;">7-CALCULE L'ÉCART-TYPE DE L'ECHANTILLON :</h3>

In [14]:
std_dev = sleep_duration.std(ddof=1)
print(f"Écart-type : {round(std_dev,2)}")

Écart-type : 0.8


<div style="background-color: #f0f8ff; padding: 10px; border-left: 5px solid #1e90ff; border-radius: 5px; color: black;">
<b>Explication :</b><br>
L'<b>écart-type</b> (0.8) est la racine carrée de la variance.<br>
Il exprime la dispersion des durées de sommeil dans la même unité (heures).<br>
La plupart des durées se situent donc entre 7.13 ± 0.8 heures (≈6.3 à 7.9 heures).
</div>
